CONTRIBUTING.md

```diff
@@ -0,0 +1,103 @@
+# Contributing to HClib
+
+Before opening a pull request,
+please be sure to read through these guidelines.
+Following the guidelines listed below will make it much more likely
+for your contributions to be quickly accepted and merged.
+
+## Licensing
+
+All contributions submitted to this project are governed
+by the terms outlined in our license agreement.
+Specifically, our license (Apache 2.0)
+has an explicit contributions clause in [section
+5](https://github.com/habanero-rice/hclib/blob/6ad778a/LICENSE#L131-L137):
+
+> Unless You explicitly state otherwise, any Contribution intentionally
+> submitted for inclusion in the Work by You to the Licensor shall be under
+> the terms and conditions of this License, without any additional terms or
+> conditions.  Notwithstanding the above, nothing herein shall supersede or
+> modify the terms of any separate license agreement you may have executed
+> with Licensor regarding such Contributions.
+
+## Commits
+
+Please follow established best practices for your commits.
+Do only one thing per commit, and write a descriptive commit message.
+[Chris Beams' *seven rules of a great Git commit
+message*](https://chris.beams.io/posts/git-commit/#seven-rules)
+provides a good set of guidelines:
+
+1. Separate subject from body with a blank line
+2. Limit the subject line to 50 characters
+3. Capitalize the subject line
+4. Do not end the subject line with a period
+5. Use the imperative mood in the subject line
+6. Wrap the body at 72 characters
+7. Use the body to explain *what* and *why* vs. *how*
+
+## Pull Requests
+
+Please follow established best practices for your pull requests.
+Give a short description of *what* your patch does in the title,
+and include a short description of *why* the patch is needed in the comment.
+A complex title or summary is a good indication that your pull request
+is doing too many things, and should be split into multiple patches.
+
+See also [*10 tips for better Pull Requests* by Mark
+Seemann](http://blog.ploeh.dk/2015/01/15/10-tips-for-better-pull-requests/).
+
+When you create a pull request, please assign one or more core team members as 
+the reviewers. You should also assign all other contributors who may be affected
+by your changes.
+
+Once you create a pull request, GitHub will automatically start a discussion 
+concerning your request. You should monitor the discussion and address the suggestions,
+concerns and critiques from other collaborators.
+
+### Merging Pull Requests
+
+Only the Core team members have the merge privileges for pull requests in the master branch, but all collaborators
+are welcome to review and discuss a particular pull request. Here's a nice set of 
+guidelines to follow when reviewing and discussing contributor's code: 
+https://github.com/thoughtbot/guides/tree/master/code-review
+
+Once a consensus has been reached among the contributors, a core member can manually perform the merge. 
```

Comment:
In past jobs, one convention I've worked with is that reviews are only integrated after 2 reviewers sign off, one of which is a "gatekeeper". A "gatekeeper" is basically an expert, in this case analogous to what we're calling a member of the "core team". Now, we don't really have non-gatekeepers at the moment; pretty much everyone pushing to this repo right now would qualify as one. So we could just say that two members of the core team are required to sign off on each review. After the first completes their review and all comments are addressed they simply leave a comment or something indicating their approval. Once the second has similarly completed a review and had all comments addressed, they do the actual merge.

The only complication for our present state is that Nick and I are the most active in terms of code changes and reviews (though most of my pushes aren't going into master), so to have a third reviewer who isn't the person putting out the review would require you (Zoran) to be responsive to regular review requests. However, if the review guidelines are followed then the result should be that those reviews don't consume large amounts of time.